tests/ztest/base/src/main.c

```diff
@@ -166,3 +166,14 @@ ZTEST(fail_in_setup, test_should_never_run)
 }
 
 ZTEST_SUITE(fail_in_setup, NULL, fail_in_setup_setup, NULL, NULL, NULL);
+
+#if CONFIG_ZTEST_STANDALONE_EXECUTION
+
+/* Standalone usage of ZTest needs the definition of the main function */
+void main(void)
+{
+	/* run the tests */
+	ztest_main();
```

Comment:
We should have a `ZTEST` here to assert that this main is the one that's running the tests.

Reply:
Agreed, this needs to assert something to count as a test.